src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -623,14 +696,39 @@ public bool HasSkipLocalsInitAttribute
         internal bool IsAutoPropertyWithGetAccessor
             => IsAutoProperty && _getMethod is object;
 
-        protected bool IsAutoProperty
-            => (_propertyFlags & Flags.IsAutoProperty) != 0;
+        internal bool IsAutoProperty
+        {
+            get
+            {
+                return (_propertyFlags & Flags.IsAutoProperty) != 0;
+            }
+        }
 
         /// <summary>
         /// Backing field for automatically implemented property, or
         /// for a property with an initializer.
         /// </summary>
-        internal SynthesizedBackingFieldSymbol BackingField { get; }
+        internal SynthesizedBackingFieldSymbol BackingField
+        {
+            get
+            {
+                EnsureAccessorsBinding();
+                return _lazyBackingFieldSymbol;
+            }
+        }
+
+        internal SynthesizedBackingFieldSymbol NonFieldKeywordBackingField
```

Comment:
>NonFieldKeywordBackingField [](http://example.com/codeflow?start=47&length=27)

It feels like this property won't be necessary for the initial scoped down PR. See also a comment for the ```BackingField``` property above.